iib/workers/tasks/build_add_deprecations.py

```diff
@@ -0,0 +1,198 @@
+            from_index_resolved,
+            temp_dir,
+            [operator_package],
+            overwrite_from_index_token,
+        )
+        if not operators_in_db:
+            log.debug(
+                '%s operator package is either not present in index or it is opted in fbc',
+                operator_package,
+            )
+            err_msg = f'Cannot add deprecations for {operator_package}'
+            log.error(err_msg)
+            raise IIBError(err_msg)
+
+        add_deprecations_to_index(
+            request_id,
+            temp_dir,
+            from_index_resolved,
+            operator_package,
+            deprecation_schema,
+            binary_image_resolved,
+            index_db_path,
+        )
+
+        _add_label_to_index(
+            'com.redhat.index.delivery.version',
+            prebuild_info['ocp_version'],
+            temp_dir,
+            'index.Dockerfile',
+        )
+
+        _add_label_to_index(
+            'com.redhat.index.delivery.distribution_scope',
+            prebuild_info['distribution_scope'],
+            temp_dir,
+            'index.Dockerfile',
+        )
+
+        arches = prebuild_info['arches']
+        for arch in sorted(arches):
+            _build_image(temp_dir, 'index.Dockerfile', request_id, arch)
+            _push_image(request_id, arch)
+
+    set_request_state(request_id, 'in_progress', 'Creating the manifest list')
+    output_pull_spec = _create_and_push_manifest_list(request_id, arches, build_tags)
+
+    _update_index_image_pull_spec(
+        output_pull_spec,
+        request_id,
+        arches,
+        from_index,
+        overwrite_from_index,
+        overwrite_from_index_token,
+        from_index_resolved,
+        add_or_rm=True,
+    )
+    _cleanup()
+    set_request_state(
+        request_id, 'complete', 'The deprecation schema was successfully added to the index image'
+    )
+
+
+def add_deprecations_to_index(
+    request_id,
+    temp_dir,
+    from_index_resolved,
+    operator_package,
+    deprecation_schema,
+    binary_image_resolved,
+    index_db_path,
+):
+    set_request_state(request_id, 'in_progress', 'Getting all deprecations present in index image')
+    from_index_configs_dir = get_catalog_dir(from_index=from_index_resolved, base_dir=temp_dir)
+    conf = get_worker_config()
+    from_index_configs_deprecations_dir = os.path.join(
+        from_index_configs_dir, conf['operator_deprecations_dir']
+    )
+    operator_dir = os.path.join(from_index_configs_deprecations_dir, operator_package)
+    if not os.path.exists(operator_dir):
+        os.makedirs(operator_dir)
+    operator_deprecations_file = os.path.join(operator_dir, f'{operator_package}.json')
+    set_request_state(request_id, 'in_progress', 'Adding deprecations to from_index')
+    with open(operator_deprecations_file, 'w') as output_file:
+        json.dump(json.loads(deprecation_schema), output_file)
```

Comment:
You probably don't need `json.loads` here, we have already established it's a valid json when we took the input from the user.

Reply:
IIRC it was adding content as string and opm validate was failing without json.loads()